src/azul/terra.py

```diff
@@ -646,19 +646,26 @@ def for_registered_user(cls, authentication: OAuth2) -> 'TDRClient':
     def drs_client(self) -> DRSClient:
         return DRSClient(http_client=self._http_client)
 
-    def get_duos(self, source: TDRSourceRef) -> Optional[MutableJSON]:
+    def get_duos(self,
+                 source: TDRSourceRef
+                 ) -> tuple[str | None, MutableJSON | None]:
         response = self._retrieve_source(source)
         try:
             duos_id = response['duosFirecloudGroup']['duosId']
         except (KeyError, TypeError):
             log.warning('No DUOS ID available for %r', source.spec)
-            return None
+            return None, None
         else:
             url = self._duos_endpoint('dataset', 'registration', duos_id)
             response = self._request('GET', url)
             if response.status == 404:
                 log.warning('No DUOS dataset registration with ID %r from %r',
                             duos_id, source.spec)
-                return None
+                return None, None
             else:
-                return self._check_response(url, response)
+                response = self._check_response(url, response)
+                assert response['consentGroups']
+                for consent_group in response['consentGroups']:
+                    assert duos_id == consent_group['datasetIdentifier'], \
+                        (duos_id, consent_group['datasetIdentifier'])
```

Comment:
You should use `require` since this is an assertion about something external. Assertions using `assert` fire if there is a bug in our code.

And, for the umpteenth time, https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#17backslashes